src/R/Components/Impl/InteractiveWorkflow/Commands/SessionInformationCommand.cs

```diff
@@ -35,12 +39,28 @@ public sealed class SessionInformationCommand : IAsyncCommand {
             return CommandResult.Executed;
         }
 
-        private void OnBrokerChanged(object sender, EventArgs e) {
+        private void OnRSessionInitialized(object sender, EventArgs e) {
             if (_interactiveWorkflow.RSession.IsRemote) {
-                PrintBrokerInformationAsync(reportTelemetry: true).DoNotWait();
+                ReplInitComplete().ContinueWith(async (t) => await PrintBrokerInformationAsync(reportTelemetry: true)).DoNotWait();
```

Comment:
Actually not sure about changing `Connected` semantics. This may be destabilizing. If we fire `Connected` when user actually can interact with REPL, we will break code that still needs to complete before allowing user to start work. There is plot manager that perform device assignments, working directory dropdown that is being populated on connect and so on.